collectors/ebpf.plugin/ebpf_shm.c

```diff
@@ -0,0 +1,507 @@
+        }
+    }
+
+    write_begin_chart(NETDATA_APPS_FAMILY, NETDATA_SHMGET_CHART);
+    for (w = root; w; w = w->next) {
+        if (unlikely(w->exposed && w->processes)) {
+            write_chart_dimension(w->name, (long long) w->shm.get);
+        }
+    }
+    write_end_chart();
+
+    write_begin_chart(NETDATA_APPS_FAMILY, NETDATA_SHMAT_CHART);
+    for (w = root; w; w = w->next) {
+        if (unlikely(w->exposed && w->processes)) {
+            write_chart_dimension(w->name, (long long) w->shm.at);
+        }
+    }
+    write_end_chart();
+
+    write_begin_chart(NETDATA_APPS_FAMILY, NETDATA_SHMDT_CHART);
+    for (w = root; w; w = w->next) {
+        if (unlikely(w->exposed && w->processes)) {
+            write_chart_dimension(w->name, (long long) w->shm.dt);
+        }
+    }
+    write_end_chart();
+
+    write_begin_chart(NETDATA_APPS_FAMILY, NETDATA_SHMCTL_CHART);
+    for (w = root; w; w = w->next) {
+        if (unlikely(w->exposed && w->processes)) {
+            write_chart_dimension(w->name, (long long) w->shm.ctl);
+        }
+    }
+    write_end_chart();
+}
+
+/**
+* Main loop for this collector.
+*/
+static void shm_collector(ebpf_module_t *em)
+{
+    shm_threads.thread = mallocz(sizeof(netdata_thread_t));
+    shm_threads.start_routine = ebpf_shm_read_hash;
+
+    netdata_thread_create(
+        shm_threads.thread,
+        shm_threads.name,
+        NETDATA_THREAD_OPTION_JOINABLE,
+        ebpf_shm_read_hash,
+        em
+    );
+
+    int apps = em->apps_charts;
+    while (!close_ebpf_plugin) {
+        pthread_mutex_lock(&collect_data_mutex);
+        pthread_cond_wait(&collect_data_cond_var, &collect_data_mutex);
+
+        if (apps) {
+            read_apps_table();
+        }
+
+        pthread_mutex_lock(&lock);
+
+        shm_send_global();
+
+        if (apps) {
+            ebpf_shm_send_apps_data(apps_groups_root_target);
+        }
+
+        pthread_mutex_unlock(&lock);
+        pthread_mutex_unlock(&collect_data_mutex);
+    }
+}
+
+/*****************************************************************
+ *  INITIALIZE THREAD
+ *****************************************************************/
+
+/**
+ * Create apps charts
+ *
+ * Call ebpf_create_chart to create the charts on apps submenu.
+ *
+ * @param em a pointer to the structure with the default values.
+ */
+void ebpf_shm_create_apps_charts(struct ebpf_module *em, void *ptr)
+{
+    UNUSED(em);
+
+    struct target *root = ptr;
+    ebpf_create_charts_on_apps(NETDATA_SHMGET_CHART,
+                               "Calls to syscall <code>shmget(2)</code>.",
+                               EBPF_COMMON_DIMENSION_CALL,
+                               NETDATA_APPS_IPC_SHM_GROUP,
+                               NETDATA_EBPF_CHART_TYPE_STACKED,
+                               20191,
+                               ebpf_algorithms[NETDATA_EBPF_INCREMENTAL_IDX],
+                               root, NETDATA_EBPF_MODULE_NAME_SHM);
```

Comment:
Please, add space between the calls for `ebpf_create_charts_on_apps`.

Reply:
Done :+1: